### PR TITLE
issue# 190 : supported writing data to BQ with description

### PR DESCRIPTION
**Pull request by himanshukohli09**

### CODE REVIEW

File: connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java [hunk 1]
```diff
@@ -264,17 +262,29 @@ void updateMetadataIfNeeded() {
...
+        newField.setDescription(marker);
+      } else if (!description.endsWith(marker)) {
+        newField.setDescription(description + " " + marker);
+      }
     }
     return newField.build();
   }
 
+  private boolean doesFieldContainsDescription(StructField field) {
+    return (field.metadata().contains("description")
```
Comment (davidrabinowitz) [inline connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java:284]: Please use the built in `StructField.getComment()`

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java (hunk 1 above)
Comment (himanshukohli09) [reply connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java:284]: Here I am reading from the key "description" of the Metadata, whereas StructField.getComment() reads from the key "comment" of the Metadata. 
So we intend to support both the keys? or just one of the keys?
 if we intend to support both the keys and if user supplies different values to both the keys what shall we do?

Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java:284]: It was my mistake to force the BigQuery name "description" to Spark, we should support the standard `getComment()`. For backwards compatibility we will support both, where the standard  `getComment()` takes precedence.

Comment (davidrabinowitz) [reply connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java:284]: Notice the comment field should also be populated when reading from BigQuery

Comment (himanshukohli09) [reply connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryIndirectDataSourceWriter.java]: Done

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

**Review by davidrabinowitz — APPROVED** [review]

File: connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java [hunk 2]
```diff
@@ -276,15 +276,21 @@ protected static Field createBigQueryColumn(StructField sparkField, int depth) {
...
+    if (description.isPresent()) fieldBuilder.setDescription(description.get());
+
+    return fieldBuilder.build();
+  }
 
-    return createBigQueryFieldBuilder(fieldName, fieldType, fieldMode, subFields)
-        .setDescription(description)
-        .build();
+  public static Optional<String> getDescriptionOrCommentOfField(StructField field) {
+    if (!field.getComment().isEmpty()) return Optional.of(field.getComment().get());
```
[the remark this answers is not in the document]
Comment (himanshukohli09) [reply connector/src/main/java/com/google/cloud/spark/bigquery/SchemaConverters.java]: Done

**Pull request merged**